File: app/src/main/java/com/jgasteiz/comics_android/helpers/GetStringResponseAsyncTask.java
package com.jgasteiz.comics_android.helpers;

import android.os.AsyncTask;
import com.jgasteiz.comics_android.interfaces.OnResponseFetched;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;


public class GetStringResponseAsyncTask extends AsyncTask<String, Void, String> {
    private OnResponseFetched mOnResponseFetched;

        private OkHttpClient client = new OkHttpClient();

        public GetStringResponseAsyncTask(OnResponseFetched onResponseFetched) {
            mOnResponseFetched = onResponseFetched;
        }

        @Override
        protected String doInBackground(String... params) {

            Request.Builder builder = new Request.Builder();
            builder.url(params[0]);
            Request request = builder.build();
            try {
                Response response = client.newCall(request).execute();

                return response.body().string();
            } catch (Exception e) {
                e.printStackTrace();
            }
            return null;
        }

        @Override
        protected void onPostExecute(String response) {
            mOnResponseFetched.callback(response);
        }
}

File: app/src/main/java/com/jgasteiz/comics_android/helpers/GetStringResponseAsyncTask.kt
package com.jgasteiz.comics_android.helpers

import android.os.AsyncTask
import com.jgasteiz.comics_android.interfaces.OnResponseFetched
import okhttp3.OkHttpClient
import okhttp3.Request


class GetStringResponseAsyncTask(private val mOnResponseFetched: OnResponseFetched) : AsyncTask<String, Void, String>() {

    private val client = OkHttpClient()

    override fun doInBackground(vararg params: String): String? {

        val builder = Request.Builder()
        builder.url(params[0])
        val request = builder.build()
        try {
            val response = client.newCall(request).execute()

            return response.body().string()
        } catch (e: Exception) {
            e.printStackTrace()
        }

        return null
    }

    override fun onPostExecute(response: String) {
        mOnResponseFetched.callback(response)
    }
}
